assets/src/components/with-term/index.js

```diff
@@ -33,30 +33,66 @@ export default function withTerm() {
 			const [ term, setTerm ] = useState( null );
 
 			useEffect( () => {
-				createTerm( taxonomy, label, value )
-					.catch( error => {
-						if ( error.code !== 'term_exists' ) {
-							return Promise.reject( error );
+				function handleSearchTerms( terms ) {
+					if ( terms.length ) {
+						setTerm( terms[ 0 ] );
+					}
+				}
+
+				function handleError( error ) {
+					if ( error.code !== 'term_exists' ) {
+						return Promise.reject( error );
+					}
+
+					return searchTerms( taxonomy, value ).then( searchTermsResponse => {
+						if ( searchTermsResponse instanceof Response ) {
+							searchTermsResponse.json().then(
+								terms => {
+									handleSearchTerms( terms );
+								}
+							);
+						} else {
+							handleSearchTerms( searchTermsResponse );
 						}
+					} );
+				}
+
+				function handeNewTerm( newTerm ) {
+					const { id, name, slug } = newTerm;
 
-						return searchTerms( taxonomy, value ).then( terms => {
-							if ( terms.length ) {
-								setTerm( terms[ 0 ] );
-							}
-						} );
+					setTerm( {
+						id,
+						name,
+						slug,
+					} );
+				}
+
+				createTerm( taxonomy, label, value )
+					.catch( response => {
+						if ( response instanceof Response ) {
+							response.json().then(
+								error => {
+									handleError( error );
```

Comment:
```suggestion
							response.json().then( handleError );
```
Same suggestion -- handler can become the callback